pkg/virt-launcher/virtwrap/network/podnic.go

```diff
@@ -134,6 +134,14 @@ func (l *podNIC) sortIPsBasedOnPrimaryIP(ipv4, ipv6 string) ([]string, error) {
 }
 
 func (l *podNIC) PlugPhase1() error {
+	podNetworkingConfigurator, err := l.newPodNetworkConfigurator()
+	if err != nil {
+		return nil
+	}
+	return l.plugPhase1WithInfraConfigurator(podNetworkingConfigurator)
+}
+
+func (l *podNIC) plugPhase1WithInfraConfigurator(podNetworkingConfigurator infraconfigurators.PodNetworkInfraConfigurator) error {
```

Comment:
I'm not too keen on this, but I got to admit it's a simple solution. I think it can be improved - what if you create the pod network configurator in the pod nic constructor, and store it as a member of the pod nic ? Similar to the dhcp config thing. 

This way, you could run the `PlugPhase1` method without issue, and would be able to set it up properly in the tests setup stage.

It would look like:

```golang
type podNIC struct {
	vmi              *v1.VirtualMachineInstance
	podInterfaceName string
	launcherPID      *int
	vmiSpecIface     *v1.Interface
	vmiSpecNetwork   *v1.Network
	handler          netdriver.NetworkHandler
	cacheFactory     cache.InterfaceCacheFactory
	dhcpConfigurator *dhcpconfigurator.Configurator
        infraConfigurator infraconfigurators.PodNetworkInfraConfigurator  // <=====
}

func newPodNIC(vmi *v1.VirtualMachineInstance, network *v1.Network, handler netdriver.NetworkHandler, cacheFactory cache.InterfaceCacheFactory, launcherPID *int) (*podNIC, error) {
	...

	infraConfigurator, err := l.newPodNetworkConfigurator() // you would have to make this a func, rather than a method 
        if err != nil { return err }

	return &podNIC{
		cacheFactory:     cacheFactory,
		handler:          handler,
		vmi:              vmi,
		vmiSpecNetwork:   network,
		podInterfaceName: podInterfaceName,
		vmiSpecIface:     correspondingNetworkIface,
		launcherPID:      launcherPID,
		dhcpConfigurator: dhcpConfigurator,
                infraConfigurator: infraConfigurator,
	}, nil
}
```

Reply:
That was my original approach, but I realize that since is only used at phase1 we will be constructing something when at phase2 that we don't use, but if this is not an issue I can go back to adding a constructor.

Reply:
I see. Good point. 

Not sure if it's not hacky, but how about you set the configurator only when `launcherPID != nil` ?

I guess it makes sense then to make the configurator a pointer.